channel/channel_test.go

```diff
@@ -0,0 +1,70 @@
+package channel
```

Comment:
In general, I would suggest splitting into smaller (unit) tests, as opposed to doing a full scenario and asserting along the way.